src/Microsoft.Extensions.Logging.AzureWebApps/Internal/WebAppsLogConfigurationReader.cs

```diff
@@ -0,0 +1,132 @@
+    /// </summary>
+    public class WebAppsLogConfigurationReader : IWebAppsLogConfigurationReader
+    {
+        private readonly IConfigurationRoot _configuration;
+        private readonly string _fileLogFolder;
+
+        private WebAppsLogConfiguration _latestConfiguration;
+        private IDisposable _changeSubscription;
+
+        /// <inheritdoc />
+        public event EventHandler<WebAppsLogConfiguration> OnConfigurationChanged;
+
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="WebAppsLogConfigurationReader"/> class
+        /// </summary>
+        /// <param name="context">The context in which the reader runs</param>
+        public WebAppsLogConfigurationReader(IWebAppsContext context)
+        {
+            if (!context.IsRunningInAzureWebApp)
+            {
+                _latestConfiguration = WebAppsLogConfiguration.Disabled;
+            }
+            else
+            {
+                _fileLogFolder = Path.Combine(context.HomeFolder, @"LogFiles\Application");
+                var settingsFile = Path.Combine(context.HomeFolder, @"Diagnostics\settings.json");
+
+                _configuration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddJsonFile(settingsFile, optional: true, reloadOnChange: true)
+                    .Build();
+
+                SubscribeToConfigurationChangeEvent();
+                ReloadConfiguration();
+            }
+        }
+
+        /// <inheritdoc />
+        public WebAppsLogConfiguration Latest
+        {
+            get
+            {
+                return _latestConfiguration;
+            }
+        }
+
+        /// <summary>
+        /// Disposes the object instance
+        /// </summary>
+        public void Dispose()
+        {
+            DisposeChangeSubscription();
+        }
+
+        private void OnConfigurationTokenChange(object state)
+        {
+            ReloadConfiguration();
+            SubscribeToConfigurationChangeEvent();
+
+            if (OnConfigurationChanged != null)
+            {
+                OnConfigurationChanged(this, _latestConfiguration);
+            }
+        }
+
+        private void SubscribeToConfigurationChangeEvent()
+        {
+            DisposeChangeSubscription();
+
+            // The token from configuration has to be renewed after each trigger
+            var changeToken = _configuration.GetReloadToken();
+            _changeSubscription = changeToken.RegisterChangeCallback(OnConfigurationTokenChange, null);
+        }
+
+        private void ReloadConfiguration()
+        {
+            // Don't use the binder because of all the defaults that we want in place
+            _latestConfiguration = new WebAppsLogConfiguration(
+                fileLoggingEnabled: TextToBoolean(_configuration.GetSection("AzureDriveEnabled")?.Value),
+                fileLoggingLevel: TextToLogLevel(_configuration.GetSection("AzureDriveTraceLevel")?.Value),
```

Comment:
AzureDriveTraceLevel => AzureDriveLogLevel ? (similarly for AzureBlobTraceLevel)